components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/impl/WorkersApiServiceImpl.java

```diff
@@ -0,0 +1,904 @@
+        // TODO: 10/16/17   This is not supported yet support with globle interceptor.
+        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "This is not supported yet"))
+                .build();
+    }
+
+    /**
+     * Read the SP Status Dashboard YML file and returen polling enterval.
+     *
+     * @return the polling interval of realtime data.
+     * @throws NotFoundException
+     */
+    @Override
+    public Response getDashboardConfig() throws NotFoundException {
+        ConfigProvider configProvider = DashboardDataHolder.getInstance().getConfigProvider();
+        DashboardConfig config = new DashboardConfig();
+        if (dashboardConfigurations != null) {
+            config.setPollingInterval(dashboardConfigurations.getPollingInterval());
+        } else {
+            loadConfig();
+            config.setPollingInterval(dashboardConfigurations.getPollingInterval());
+        }
+        String jsonString = new Gson().toJson(config);
+        return Response.ok().entity(jsonString).build();
+    }
+
+    private String getAdminUsername() {
+        if (dashboardConfigurations != null) {
+            return dashboardConfigurations.getAdminUsername();
+        } else {
+            loadConfig();
+            return dashboardConfigurations.getAdminUsername();
+        }
+    }
+
+    private String getAdminPassword() {
+        if (dashboardConfigurations != null) {
+            return dashboardConfigurations.getAdminPassword();
+        } else {
+            loadConfig();
+            return dashboardConfigurations.getAdminPassword();
+        }
+    }
+
+    private void loadConfig() {
+        ConfigProvider configProvider = DashboardDataHolder.getInstance().getConfigProvider();
+        DashboardConfig config = new DashboardConfig();
+        try {
+            dashboardConfigurations = configProvider
+                    .getConfigurationObject(SpDashboardConfiguration.class);
+        } catch (ConfigurationException e) {
+            logger.error("Error getting the dashboard configuration.", e);
+        }
+    }
+
+    private long parsPeriod(String interval) {
+        long millisVal = DEFAULT_TIME_INTERVAL_MILLIS;
+        String numberOnly = interval.replaceAll("[^0-9]", "");
+        if (interval.contains("sec")) {
+            millisVal = Long.parseLong(numberOnly) * 1000;
+        } else if (interval.contains("min")) {
+            millisVal = Long.parseLong(numberOnly) * 60000;
+        } else if (interval.contains("hr")) {
+            millisVal = Long.parseLong(numberOnly) * 3600000;
+        } else if (interval.contains("ms")) {
+            millisVal = Long.parseLong(numberOnly);
+        } else {
+            try {
+                millisVal = Long.parseLong(interval);
+            } catch (ClassCastException | NumberFormatException e) {
+                logger.error("Invalid parsing the value time period " + interval + " to milliseconds. Hence proceed " +
```

Comment:
use String.format() to concatenate strings. Add stackTrace to the log.
Please make update the other places also accordingly.